web/worker_service/worker_service.js

```diff
@@ -0,0 +1,139 @@
+          // When the app updates the serviceWorkerVersion changes, so we
+          // need to ask the service worker to update.
+          console.log('[TwakeMail] fetchServiceWorker(): New service worker available.');
+          reg.update();
+          waitForActivation(reg.installing);
+        } else {
+          // Existing service worker is still good.
+          console.log('[TwakeMail] fetchServiceWorker(): Loading app from service worker.');
+          loadMainDartJs();
+        }
+      });
+    // If service worker doesn't succeed in a reasonable amount of time,
+    // fallback to plaint <script> tag.
+    setTimeout(() => {
+      if (!scriptLoaded) {
+        console.warn(
+          '[TwakeMail] fetchServiceWorker(): Failed to load app from service worker. Falling back to plain <script> tag.',
+        );
+        loadMainDartJs();
+      }
+    }, timeoutDuration);
+  } else {
+    // Service workers not supported. Just drop the <script> tag.
+    loadMainDartJs();
+  }
+}
+
+function handleContinueTwakeMailOnWeb() {
+  console.info('[TwakeMail] handleContinueTwakeMailOnWeb(): Continue on web.');
+  closeSmartBanner();
+  fetchServiceWorker();
+}
+
+function getPlatform() {
+  console.info('[TwakeMail] getPlatform(): ', navigator.userAgent);
+  if (/iPhone|iPad|iPod/i.test(navigator.userAgent)) {
+    return iosPlatform;
+  }
+  if (/Android/i.test(navigator.userAgent)) {
+    return androidPlatform;
+  }
+  return otherPlatform;
+}
+
+function handleOpenTwakeMailApp() {
+  const os = getPlatform();
+  console.info('[TwakeMail] handleOpenTwakeMailApp() - OS:', os);
+  if (os === androidPlatform) {
+    document.location.replace(androidStore);
+  } else if (os === iosPlatform) {
+    document.location.replace(iosStore);
+  }
+}
+
+function initialWorkerService() {
+  const os = getPlatform();
+  const originInUrl = window.location;
+
+  console.info('[TwakeMail] initialWorkerService(): OriginInUrl:', originInUrl);
+
+  //   For desktop, we don't show the open on app popup
+  if (os === otherPlatform || typeof window === 'undefined') {
+    fetchServiceWorker();
+    return;
+  }
+
+  if (window.location.pathname.includes('/login')) {
+    console.info('[TwakeMail] initialWorkerService(): Login callback');
+    handleContinueTwakeMailOnWeb();
+  } else {
+    openSmartBanner();
+  }
+}
+
+function openSmartBanner() {
+  console.info('[TwakeMail] openSmartBanner(): Open the smart banner.');
+  const smartBanner = document.querySelector(".smart-banner");
+  smartBanner.style.display = "block";
+  document.body.style.overflow = "hidden";
+  smartBanner.style.top = "16px";
+}
+
+function closeSmartBanner() {
+  console.info('[TwakeMail] closeSmartBanner(): Closing the smart banner.');
+  const smartBanner = document.querySelector(".smart-banner");
+  smartBanner.style.display = "none";
+  document.body.style.overflow = "auto";
+  smartBanner.style.bottom = "-100%";
```

Comment:
Banner open base on top attribute. So we should set `bottomSheet.style.top` back to 0 here